File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        
        # store value -> index
        for index, value in enumerate(nums):
            mp[value] = index
        
        # check for complement
        for ind, val in enumerate(nums):
            complement = target - val
            if complement in mp and mp[complement] != ind:
                return [ind, mp[complement]]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
       unordered_map<int,int>mpp;
       for(int i=0;i<nums.size();i++){
         mpp[nums[i]]=i;
       }
       
    for(int i=0;i<nums.size();i++){
        if(mpp.find(target-nums[i])!=mpp.end()){
         if(i==mpp[target-nums[i]])
             continue;
             
         if(i<mpp[target-nums[i]]){
            return {i,mpp[target-nums[i]]};
         }  
         else{
            return {mpp[target-nums[i]],i};
         }
        
        }
    }
 return {-1,-1};

    }
};